modules/nextflow/src/main/groovy/nextflow/spack/SpackCache.groovy

```diff
@@ -332,21 +339,40 @@ class SpackCache {
      *      The {@link DataflowVariable} which hold (and pull) the local image file
      */
     @PackageScope
-    DataflowVariable<Path> getLazyImagePath(String spackEnv) {
+    DataflowVariable<Path> getLazyImagePath(String spackEnv, String arch) {
+
+        def spackEnvArch
+
+        if ( arch ) {
+            spackEnvArch = spackEnv + "_" + arch
+        }
+        else {
+            spackEnvArch = spackEnv
+        }
```

Comment:
A conditional expression can make this more concise e.g. 

```
spackEnvArch = arch ? spackEnv + "_" + arch : spackEnv
```

Reply:
adding it